pkg/ignore/ignore.go

```diff
@@ -1,66 +1,101 @@
 package ignore
 
 import (
-	"errors"
-	"io/ioutil"
+	"fmt"
 	"os"
 	"strings"
 	"time"
 
-	gitignore "github.com/get-woke/go-gitignore"
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/osfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/format/gitignore"
 	"github.com/rs/zerolog/log"
+
+	"github.com/get-woke/woke/pkg/util"
 )
 
 // Ignore is a gitignore-style object to ignore files/directories
 type Ignore struct {
-	matcher *gitignore.GitIgnore
+	matcher gitignore.Matcher
 }
 
-var defaultIgnoreFiles = []string{
-	".gitignore",
-	".ignore",
-	".wokeignore",
-	".git/info/exclude",
+// given a absolute path (example: /runner/folder/root/data/effx.yaml)
+// and given a workingDir (example: root)
+// it will return data/effx.yaml
+func getDomainFromWorkingDir(absPath, workingDir string) []string {
+	// if working directory does not end with a slash, add it
+	if strings.LastIndex(workingDir, string(os.PathSeparator)) != len(workingDir)-1 {
+		workingDir += string(os.PathSeparator)
+	}
+
+	res := strings.Split(absPath, workingDir)
+	if len(res) > 1 {
+		return util.FilterEmptyStrings(strings.Split(res[1], string(os.PathSeparator)))
+	}
+	return []string{}
 }
 
 // NewIgnore produces an Ignore object, with compiled lines from defaultIgnoreFiles
 // which you can match files against
-func NewIgnore(lines []string) *Ignore {
+func NewIgnore(lines []string, findRootDir bool) *Ignore {
 	start := time.Now()
 	defer func() {
 		log.Debug().
 			TimeDiff("durationMS", time.Now(), start).
 			Msg("finished compiling ignores")
 	}()
 
-	for _, filename := range defaultIgnoreFiles {
-		lines = append(lines, readIgnoreFile(filename)...)
+	cwd, err := os.Getwd()
+	if err != nil {
+		fmt.Println(err)
+		return nil
 	}
 
-	ignorer := Ignore{
-		matcher: gitignore.CompileIgnoreLines(lines...),
+	// DetectDotGit will traverse parent directories until it finds the root git dir if true
+	opt := &git.PlainOpenOptions{DetectDotGit: findRootDir}
+	repo, err := git.PlainOpenWithOptions("", opt)
```

Comment:
Is there ever a chance that `repo` and `err` are both `nil`? It might be worth guarding against `filesystem` never being set